metrics: Support Trend.Max with negative values

codebien opened the pull request:
Max is not computed correctly when negative numbers are used.

<!--
  (ﾉ◕ヮ◕)ﾉ*:・ﾟ✧
  
  Thank you for your interest in contributing to the k6 project!
  
  Before you get started, we'd kindly like to ask you to read our:
    - Contribution guidelines at https://github.com/grafana/k6/blob/master/CONTRIBUTING.md
    - Code of Conduct at https://github.com/grafana/k6/blob/master/CODE_OF_CONDUCT.md
    
  Out of respect for your time, please start a discussion regarding any bigger contributions either
  in a GitHub Issue, in the community forums or in the #contributors channel of the k6 slack before you
  get started on the implementation.
  
  If you've already done all of that, you're more than welcome to proceed with your pull request.
  Thank you again for your contribution! 🙏🏼
  
  
-->

mstoykov commented on metrics/sink.go:
```diff
@@ -86,10 +86,14 @@ func (t *TrendSink) Add(s Sample) {
 	t.Sum += s.Value
 	t.Avg = t.Sum / float64(t.Count)
 
+	if t.Count == 1 {
+		t.Max, t.Min = s.Value, s.Value
+		return
+	}
```
nit: I feel this would be kind of nicer if it is `if t.Count ==0` at the very beginning with everything that happens on the *first* `Add` .

And then have the rest as is.

*Alternatively* we could have TrendSink be created with `NewTrendSink` which sets Max,Min to `-math.MaxFloat64` :thinking: this will make this logic not needed on each `Add`

olegbespalov replied:
The idea of having a proper constructur `NewTrendSink` sounds good :+1:

codebien replied:
I also like this proposal, but I'm against doing the change in this PR (essentially in this release) for the following reasons:
1. We already have the Sink API under discussion at https://github.com/grafana/k6/pull/2898 and I would like to get a consensus there before. I need to restart that work.
2. I would like to have consistent Sink implementations so it would require adding the constructors also for the other sinks and it seems to me that it is too much for a last-minute change.

I add it as a suggestion to #2898 and will address it there.

olegbespalov replied:
Make sense! Thanks for the explanation :+1:

mstoykov reviewed (approved):
LGTM!

Left a comment with some alternatives to the current solution

olegbespalov commented on metrics/sink_test.go:
```diff
@@ -93,24 +99,43 @@ func TestTrendSink(t *testing.T) {
 			assert.Equal(t, 100.0, sink.Max)
 			assert.Equal(t, 54.0, sink.Avg)
 		})
+		t.Run("negative", func(t *testing.T) {
```
Is it possible to have a "mixed" Trend, e.g., `-1,0,2`? If so, is itng a test case for it is good?

codebien commented:
Moved the condition at the beginning so `t.Count == 0 {` can be used. Added the checks for Values and Sum in the tests and removed sorted that is an internal field.

For Sum/Sub operations we should define a precision we would like to have to have better numbers. (Sum is not used in k6 so it is out of scope here).
